src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -168,28 +176,19 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
         {
```

Comment:
We wait on this task to get the socket in the first place. ConnectAsync is a big async method.